operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -233,8 +238,27 @@ func extractGCSConfigSecret(s *corev1.Secret) (*storage.GCSStorageConfig, error)
 		return nil, fmt.Errorf("%w: %s", errSecretMissingField, storage.KeyGCPServiceAccountKeyFilename)
 	}
 
+	credentialsFile := struct {
+		CredentialsType string `json:"type"`
+	}{}
```

Comment:
I have seen, that the `.credential_source.file` needs to match the VolumeMount we create in our configuration. This bothers me, because it makes the user configuration dependent on our internal configuration. It would be nice if there was a possibility to override this (for example an environment variable), but so far I have not found anything in the configuration.

If the contents of the secret need to match our configuration then I think we need to validate this. An alternative, albeit more complex, would be to write our own configuration based on what the user provides.